ruff_lsp/server.py

```diff
@@ -588,46 +595,76 @@ def apply_organize_imports(arguments: tuple[TextDocument]):
     uri = arguments[0]["uri"]
     text_document = LSP_SERVER.workspace.get_document(uri)
     LSP_SERVER.apply_edit(
-        _create_workspace_edits(
-            text_document, _formatting_helper(text_document, only="I001") or []
-        ),
+        _create_workspace_edits(text_document, _fix_helper(text_document, only="I001")),
         "Ruff: Format imports",
     )
 
 
-def _formatting_helper(
-    document: workspace.Document, *, only: str | None = None
+if os.environ.get("RUFF_BETA_INTERNAL"):
+
+    @LSP_SERVER.command("ruff.applyFormat")
+    def apply_format(ls: server.LanguageServer, arguments: tuple[TextDocument]):
+        apply_format_impl(ls, arguments)
+
+
+def apply_format_impl(ls: server.LanguageServer, arguments: tuple[TextDocument]):
+    uri = arguments[0]["uri"]
+    document = ls.workspace.get_document(uri)
+    log_to_output(f"Formatting {uri}")
+    result = _run_subcommand_on_document(
+        document, args=["format", "-"], source=document.source
+    )
+    edits = _result_to_edits(document, result)
+    ls.apply_edit(
+        _create_workspace_edits(document, edits),
+        "Ruff: Format code",
+    )
+
+
+def _fix_helper(
+    document: workspace.Document,
+    *,
```

Comment:
What's the meaning of the `*`?

Reply:
Afterwards keyword argument only, e.g. `_fix_helper(document, only="F")` is valid but `_fix_helper(document, "F")` is not. This part i haven't change, no specific opinion given that those are internal APIs and we can change them at will.

https://stackoverflow.com/questions/59661042/what-do-single-star-and-slash-do-as-independent-parameters